janusgraph-core/src/main/java/org/janusgraph/graphdb/database/management/ManagementSystem.java

```diff
@@ -910,22 +907,39 @@ public ScanJobFuture updateIndex(Index index, SchemaAction updateAction, int num
                 future = new EmptyScanJobFuture();
                 break;
             case DISABLE_INDEX:
-                setStatus(schemaVertex, SchemaStatus.INSTALLED, keySubset);
+                setStatus(schemaVertex, SchemaStatus.DISABLED, keySubset);
```

Comment:
Do you know why the old code set the schema vertex to `INSTALLED` first and then set an update trigger, and why we don't do it here?

Reply:
I noticed that and failed to come up with reasons why it would be beneficial to set the state to `INSTALLED` first. A little bit more background:

- `setStatus` modifies the schema vertex to update the status property. This happens within a transaction and is not written to the storage backend until the management transaction is committed.
- The update trigger is applied during the commit phase of a management transaction. Therefore in the old version, the status is set to `INSTALLED` and immediately overwritten to `DISABLED` when committing the management transaction.

Consequently, the only difference is the state in which we see the index within our uncommitted transaction. I can only speculate about the intended effects of this:
- Maybe the authors wanted to prevent a subsequent call to `REMOVE_INDEX` within the same transaction. That would have been prevented because we see the index as `INSTALLED` after calling `DISABLE_INDEX` until the transaction is committed.
- That, however, would introduce the ability to erroneously call `REGISTER_INDEX` as long as the transaction is still open and have the state end up in `REGISTERED`, which is unexpected behavior and should not be possible according to the index lifecycle.

Both of these arguments still apply to the updated version in a similar way. Instead of exchanging one possibly unwanted effect for another, I opted for readability of the code.

Reply:
That makes sense, thanks for the detailed explanation!